jdi-light-bootstrap-tests/src/test/java/io/github/epam/bootstrap/tests/composite/section/modal/ModalVerticallyCenteredTests.java

```diff
@@ -0,0 +1,168 @@
+                        modalVerticallyCentered.dismissModal1Close,
+                        "button:nth-of-type(1)",
+                        "modal-vertical-content-1"
+                },
+
+                {
+                        modalVerticallyCentered.modalCenterScrollableTrigger,
+                        modalVerticallyCentered.dismissModal2Close,
+                        "button:nth-of-type(2)",
+                        "modal-vertical-content-2"
+                }
+        };
+    }
+
+    @DataProvider
+    public Object[][] modalDismissData() {
+        return new Object[][]{
+                {
+                        modalVerticallyCentered.modalCenterTrigger,
+                        modalVerticallyCentered.dismissModal1Close,
+                        modalVerticallyCentered.closeX
+                },
+
+                {
+                        modalVerticallyCentered.modalCenterScrollableTrigger,
+                        modalVerticallyCentered.dismissModal2Close,
+                        modalVerticallyCentered.closeX
+                }
+        };
+    }
+
+    @BeforeMethod
+    public void before() {
+        shouldBeLoggedIn();
+        bsPage.shouldBeOpened();
+    }
+
+    @Test(dataProvider = "modalBasicData")
+    public void modalBasicFunctionalityTest(Button showButton,
+                                            Button dismissButton,
+                                            Modal modal) {
+        showButton.show();
+        showButton.click();
+
+        modal.is().displayed();
+
+        dismissButton.show();
+        dismissButton.click();
+
+        modal.is().hidden();
+    }
+
+    @Test(dataProvider = "modalCssData")
+    public void modalCssTest(Button showButton,
+                             Button dismissButton,
+                             Modal modal,
+                             String modalBgCss) {
+        showButton.show();
+        showButton.click();
+
+        modal.core().hasClass(modalBgCss);
+
+        dismissButton.show();
+        dismissButton.click();
+    }
+
+    @Test(dataProvider = "modalVerticalAlignmentData")
+    public void modalVerticalAlignmentTest(Button showButton,
+                                           Button dismissButton,
+                                           String waiterSelector,
+                                           String modalId) {
+        showButton.show();
+        showButton.click();
+
+        $(waiterSelector).waitFor().hidden();
+
+        long modalTop = WebDriverFactory.jsExecute(
+                "var modal = document.getElementById('" + modalId + "');" +
+                        "return modal.getBoundingClientRect().top;"
+        );
+        long modalBottom = WebDriverFactory.jsExecute(
+                "var modal = document.getElementById('" + modalId + "');" +
+                        "return window.innerHeight - modal.getBoundingClientRect().bottom;"
+        );
+
+        assertEquals(modalTop, modalBottom);
```

Comment:
it's better to create jdi assertion to compare these two values instead of raw using of testng assertions

Reply:
Refactored via **assertThat()**